closed/src/java.base/share/classes/openj9/internal/security/RestrictedSecurity.java

```diff
@@ -0,0 +1,1053 @@
+                        printStackTraceAndExit("Constraint format is incorrect: " + providerInfo);
+                    }
+
+                    // Constraint object array.
+                    // For each constraint type, algorithm and attributes.
+                    Constraint[] constraints = new Constraint[constrArray.length];
+
+                    int cNum = 0;
+                    for (String constr : constrArray) {
+                        String[] input = constr.split(",");
+
+                        // Each constraint must includes 3 fields(type, algorithm, attributes).
+                        if (input.length != 3) {
+                            printStackTraceAndExit("Constraint format is incorrect: " + providerInfo);
+                        }
+
+                        String inType = input[0].trim();
+                        String inAlgorithm = input[1].trim();
+                        String inAttributes = input[2].trim();
+
+                        // Each attribute must includes 2 fields(key and value) or *.
+                        if (!isAsterisk(inAttributes)) {
+                            String[] attributeArray = inAttributes.split(":");
+                            for (String attribute : attributeArray) {
+                                String[] in = attribute.split("=", 2);
+                                if (in.length != 2) {
+                                    printStackTraceAndExit(
+                                            "Constraint attributes format is incorrect: " + providerInfo);
+                                }
+                            }
+                        }
+
+                        Constraint constraint = new Constraint(inType, inAlgorithm, inAttributes);
+
+                        if (debug != null) {
+                            debug.println("Loading constraints for security provider: " + providerName
+                                    + " with constraints type: " + constraint.type
+                                    + " algorithm: " + constraint.algorithm
+                                    + " attributes: " + constraint.attributes);
+                        }
+                        constraints[cNum] = constraint;
+                        cNum++;
+                    }
+                    providerConstraints.put(providerName, constraints);
+                    if (debug != null) {
+                        debug.println("Loaded constraints for security provider: " + providerName);
+                    }
+                } else {
+                    printStackTraceAndExit("Constraint format is incorrect: " + providerInfo);
+                }
+            }
+        }
+
+        /**
+         * Check if the Service is allowed in restricted security mode.
+         *
+         * @param service the Service to check
+         * @return true if the Service is allowed
+         */
+        private boolean isRestrictedServiceAllowed(Service service) {
```

Comment:
`private` should only be used for features which are only used from within the class.